tests/test_swift.py

```diff
@@ -87,35 +73,139 @@ def setUp(self):
         self.swift = swift.Swift(APP_INFOS)
 
     def test_init(self):
+        self.assertEqual(self.swift.appInfos, APP_INFOS)
         self.assertIsInstance(self.swift.mainWindow, QMainWindow)
         self.assertIsInstance(self.swift.centralWidget, QLabel)
         for name in APP_INFOS:
+            self.assertIn(name, self.swift._dockWidgets)
             self.assertIn(name, self.swift._apps)
         for channel in self.channels:
             self.assertIn(channel, self.swift._subscribers)
 
+    def test_app_names(self):
+        appNames = self.swift.appNames()
+        for name in APP_INFOS:
+            self.assertIn(name, appNames)
+
     def test_destroy_app(self):
         for name in APP_INFOS:
             self.swift.destroyApp(name)
             self.assertNotIn(name, self.swift._apps)
 
+    def test_update_frames(self):
+        self.swift._apps["app1"].frames.return_value = (QWidget(),)
+        self.swift.updateFrames("app1")
```

Comment:
Line 96 means that `frames()` of the app is updated because `QWidget()` is different from the original frame.
So I checked if `updateFrames()` runs correctly.

Howerver, as you said, the test cases seem to be unsufficient. I will update it.

Reply:
Wow! It was a too important review. 
`Swift.removeFrame()` didn't run correctly because it didn't remove the frame from `self.swift._dockwidgets`.